cirq/ops/controlled_test_util.py

```diff
@@ -0,0 +1,54 @@
+from typing import Union, Tuple, cast
+
+import numpy as np
+import cirq
+from cirq.type_workarounds import NotImplementedType
+
+
+class GateUsingWorkspaceForApplyUnitary(cirq.SingleQubitGate):
```

Comment:
Move these into the main test class adjacent to the test they're used in. I've had bad past experiences with "common test utilities" exploding into a nightmare of interdependency.

Reply:
These are used in both tests - for controlled op and controlled gate - are we okay with the duplication?

Reply:
I created the duplicated version - do you prefer this? or am I missing some pytest specific structure?